changelogs/unreleased/gh-5272-sync-timeout-default.md

```diff
@@ -0,0 +1,13 @@
+## feature/replication
+
+* Added a compat option "box_cfg_replication_sync_timeout" to control the
+  default value of "replication_sync_timeout" box.cfg option. Old default is 300
+  seconds and new default is 0. The noticeable difference in new behavior is
+  that `box.cfg{replication = ""}` call returns before the node is synced with
+  remote instances. If one expects the node to be writable once `box.cfg`
+  returns, this can be achieved with new behavior by calling `box.ctl.wait_rw()`
+  after `box.cfg{replication=...}`.
```

Comment:
I don't know, whether it is important, but is there a workaround for `box.cfg({read_only = true, replication = <...>})`?

Reply:
I suppose no. The main difference is whether the node is writable or not after `box.cfg{}` call returns.
If the config contains `read_only = true`, the node won't be writable anyway.

Reply:
It seems, it is not a question for those changelog/compat option changes, but rather about the change itself, but I'll summarize it just in case.

Let's assume I have a read only node, which serves read requests (to reduce load on writeable nodes). Technically I should handle a possibility to move into an orphan state due to a network problems to don't feed stale data to clients (and should also handle large replication lags). However, if I have a good network, it is possible that I neven meet this situation and don't handle it carefully.

After the change I'll meet the situation after a restart with a good probability. So, technically, there should be a way to fail iproto requests, when the data is stale (orphan mode or large replication lag). Preferably as a box option, because it is simpler than a middleware for all iproto functions and triggers for all spaces. I would also add an option to the http server to fail http requests in this case.

However it is definitely out of scope here. I'm closing the discussion.